src/coreclr/jit/gentree.cpp

```diff
@@ -21295,19 +21295,104 @@ GenTree* Compiler::gtNewSimdBinOpNode(
 #if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
         case GT_DIV:
         {
-            if (simdBaseType == TYP_INT)
+            if (varTypeIsIntegral(simdBaseType))
             {
-                assert(compIsaSupportedDebugOnly(InstructionSet_AVX) ||
-                       compIsaSupportedDebugOnly(InstructionSet_AVX512));
+                assert(!varTypeIsLong(simdBaseType));
+                if (((varTypeIsShort(simdBaseType) || varTypeIsByte(simdBaseType)) && simdSize > 16) ||
+                    (!compOpportunisticallyDependsOn(InstructionSet_AVX512) &&
+                     (simdSize > 32 || (varTypeIsInt(simdBaseType) && simdSize == 32))) ||
+                    (!compOpportunisticallyDependsOn(InstructionSet_AVX) && simdSize > 16) || simdSize == 64)
+                {
+                    var_types divType  = simdSize == 64 ? TYP_SIMD32 : TYP_SIMD16;
+                    GenTree*  op1Dup   = fgMakeMultiUse(&op1);
+                    GenTree*  op2Dup   = fgMakeMultiUse(&op2);
+                    GenTree*  op1Lower = gtNewSimdGetLowerNode(divType, op1, simdBaseJitType, simdSize);
+                    GenTree*  op2Lower = gtNewSimdGetLowerNode(divType, op2, simdBaseJitType, simdSize);
+                    GenTree*  divLower =
+                        gtNewSimdBinOpNode(GT_DIV, divType, op1Lower, op2Lower, simdBaseJitType, simdSize / 2);
+                    GenTree* op1Upper = gtNewSimdGetUpperNode(divType, op1Dup, simdBaseJitType, simdSize);
+                    GenTree* op2Upper = gtNewSimdGetUpperNode(divType, op2Dup, simdBaseJitType, simdSize);
+                    GenTree* divUpper =
+                        gtNewSimdBinOpNode(GT_DIV, divType, op1Upper, op2Upper, simdBaseJitType, simdSize / 2);
+                    GenTree* divResult = gtNewSimdWithUpperNode(type, divLower, divUpper, simdBaseJitType, simdSize);
+                    return divResult;
+                }
+
+                if (varTypeIsShort(simdBaseType) || varTypeIsByte(simdBaseType))
```

Comment:
nit: This could also be
```suggestion
                if (varTypeIsSmall(simdBaseType))
```

Reply:
Updated to `varTypeIsSmall`